### PR TITLE
gnome-shell, gnome-tweak-tool: Don't propagate python

### PR DESCRIPTION
**Pull request by jgeerds**
###### Motivation for this change
see https://github.com/NixOS/nixpkgs/issues/21851 for more details.

I don't think it is a good idea to propagate python in `gnome3.gnome-shell` because this leads to issues like in `gnome3.gnome-tweak-tools` (multiple python versions in build env).

The only reason I see for propagating is that `gnome-shell-extension-tool` didn't work in the past. My changes fixes `gnome-shell-extension-tool` without propagating python.

I've also removed the propagated python dependency in `gnome-tweak-tools` because... why not? It still works for me and it is IHMO less error prone.

Are there any downsides? If not we could also remove the `propagatedBuildInputs` in other (gnome specific) derivations.

###### Things done

- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/desktops/gnome-3/3.22/misc/gnome-tweak-tool/default.nix [hunk 1]
```diff
@@ -4,35 +4,25 @@
...
   buildInputs = [ pkgconfig gtk3 glib intltool itstool libxml2
                   gnome3.gsettings_desktop_schemas makeWrapper file
                   gdk_pixbuf gnome3.defaultIconTheme librsvg
                   libnotify gnome3.gnome_shell
                   libsoup gnome3.gnome_settings_daemon gnome3.nautilus
-                  gnome3.gnome_desktop wrapGAppsHook ];
-
-  propagatedBuildInputs = [ python gobjectIntrospection ];
+                  gnome3.gnome_desktop wrapGAppsHook
+                  python2Packages.pygobject3.dev pythonEnv gobjectIntrospection ];
```
Comment (jgeerds) [inline pkgs/desktops/gnome-3/3.22/misc/gnome-tweak-tool/default.nix]: Extra `python2Packages.pygobject3.dev` is needed for `pygobject.h`:

```
nix-build -A python2Packages.pygobject3.dev

./result-dev
├── include
│   └── pygobject-3.0
│       └── pygobject.h
├── lib
│   └── pkgconfig
│       └── pygobject-3.0.pc

```

**Comment by jgeerds** [comment]
CC: @dbeecham @dezgeg @FRidh @eqyiel

**Comment by FRidh** [comment]
I recall having looked at this in the past. If it works, then its fine with me.

**Comment by dbeecham** [comment]
Propagating anything seems to me like a bad idea. Maybe propagating build inputs solved some issue in the past, not sure, but I'd definitely like for builds to be as predictable as possible; propagating build inputs is too invisible for my taste.

**Pull request merged**